packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/widget/SelectWidget.tsx

```diff
@@ -0,0 +1,112 @@
+import { Widget, WidgetProps, EnumOptionsType } from '@rjsf/utils'
+import React from 'react'
+import { Autocomplete, TextField } from '@mui/material'
+import { createFilterOptions } from '@mui/material/Autocomplete'
+import { isObject } from 'lodash-es'
+
+const filter = createFilterOptions<EnumOptionsType>()
+
+/**
+ * This function finds the enum option that has the corresponding value and returns it.
+ *
+ * If value is nullish, this fn returns undefined. If there is no corresponding option, then a new object is returned
+ * where the label and value are set to the provided value.
+ */
+export function findValueOption(
+  value: string | null | undefined,
+  options: EnumOptionsType[],
+): EnumOptionsType | undefined {
+  if (value == null) {
+    return undefined
+  }
+
+  const correspondingOption = options.filter(option => option.value === value)
+  if (correspondingOption.length > 0) {
+    return correspondingOption[0]
+  } else {
+    return { value: value, label: value }
+  }
+}
+
+type SelectWidgetProps = WidgetProps & {
+  isClearable?: boolean
+}
+
+/**
+ * Select widget compatible with react-jsonschema-form enumerations
+ */
+export const SelectWidget: Widget = (props: SelectWidgetProps) => {
+  const {
+    id,
+    options,
+    value,
+    required,
+    disabled,
+    readonly,
+    multiple,
+    autofocus,
+    onChange,
+    onBlur,
+    onFocus,
+    placeholder,
+    isClearable = true,
+    rawErrors,
+  } = props
+  const { enumOptions } = options as {
+    enumOptions: EnumOptionsType[]
+  }
+  return (
+    <Autocomplete
+      id={id}
+      value={findValueOption(value, enumOptions)}
+      freeSolo
+      forcePopupIcon
+      selectOnFocus
+      clearOnBlur
+      handleHomeEndKeys
+      options={enumOptions}
+      readOnly={readonly}
+      disabled={disabled}
+      multiple={multiple}
+      disableClearable={!isClearable}
+      onChange={(event, newValue) => {
+        if (isObject(newValue) && 'inputValue' in newValue) {
+          // Create a new value from the user input
+          onChange(newValue.inputValue)
+        } else if (isObject(newValue) && 'value' in newValue) {
+          onChange(newValue.value)
+        } else if (typeof newValue === 'string') {
+          onChange(newValue)
+        } else {
+          onChange(newValue)
```

Comment:
Does `onChange` handle other cases gracefully?  Are there cases when `newValue` is an object but doesn't have an `inputValue` or `value` property?

Reply:
Not totally sure if this is possible, MUI seems to handle fallback like this in their [Creatable example](https://mui.com/material-ui/react-autocomplete/#creatable)